### PR TITLE
Issue 2734: Fix formatting errors in documentation.

### LINKED ISSUE #2734: Formatting issues in mark down documentation
**Problem description**
There are few issues in the format of the text, when they are converted to html, though they look perfectly fine when we look at the actual `.md` file. 
for ex : 
1. `*` are not always converted to bullet point in `html`, we need to use `-`.
2. The images in `controller-service.md` are not rendered properly when converted to html
3. github relative links works fine, when the reference is made to same folder, but when we refer to root folder for ex : if we refer to `README.md` from documentation using a relative path, it works only in `github` not in pravega.io website.

**Problem location**
documentation/src/docs/

**Suggestions for an improvement**

### PR DESCRIPTION
**Pull request by Prabha-Veerubhotla**
**Change log description**
* Fixes format of bullet points everywhere
* Properly renders images in controller-service documentation.
* Corrects the link for state-synchronizer-design from reader group design doc.
* Fixes code format in add own metrics section and also format of available metrics.

**Purpose of the change**
Fixes #2734 

**What the code does**
Some of the documentation got into formatting issues, after conversion to html docs, when the `v0.3.0` docs are generated. 

**How to verify it**
No code changes, The text format should be same in HTML as well as in markdown format.

### CODE REVIEW

**Comment by Prabha-Veerubhotla** [comment]
Note:  **The documents are already updated in proper format in the pravega.io website.**

**Comment by RaulGracia** [comment]
I found a couple of markdown errors in `http://pravega.io/docs/latest/segment-store-service`: There are two places where `_Durable Data Log` should be `_Durable Data Log_` (to correctly format as italics).

**Comment by Prabha-Veerubhotla** [comment]
> I found a couple of markdown errors in http://pravega.io/docs/latest/segment-store-service: There are two places where _Durable Data Log should be _Durable Data Log_ (to correctly format as italics).

fixed.

File: documentation/src/docs/pravega-concepts.md [hunk 1]
```diff
@@ -146,7 +146,7 @@ space is then divided into a number of partitions, corresponding to the number
 of Stream Segments. Consistent hashing determines which Segment an Event is
 assigned to.
 
-### AutoScaling: The number of Stream Segments can vary over time
+### AutoScaling: Number of Stream Segments can vary over time
```
Comment (fpj) [inline documentation/src/docs/pravega-concepts.md]: Revert this change, please.

File: documentation/src/docs/getting-started.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ Lets download and run the "Hello World" Pravega sample reader and writer apps. P
 dependencies will be pulled from maven central.
 
 Note: The samples can also use a locally compiled version of Pravega. For more information
-about this see the note on maven publishing in the [README.md](../../../README.md).
+about this see the note on maven publishing in the [README.md](https://github.com/pravega/pravega/blob/master/README.md).
```
Comment (fpj) [inline documentation/src/docs/getting-started.md]: If this is about samples, then why is it pointing to `pravega/pravega`? It should be pointing to `pravega/pravega-samples`.

File: documentation/src/docs/getting-started.md [hunk 3]
```diff
@@ -105,5 +105,5 @@ No more events from examples/helloStream
 ...
 ```
 
-See the readme.md file in the pravega-client-examples for more details on running the
+See the [readme.md](https://github.com/pravega/pravega-samples/blob/v0.3.0/pravega-client-examples/README.md) file in the pravega-client-examples for more details on running the
```
Comment (fpj) [inline documentation/src/docs/getting-started.md]: `README.md` capital to be consistent with the previous ones.

File: documentation/src/docs/streamcuts.md [hunk 4]
```diff
@@ -92,4 +92,4 @@ It must be noted that passing ```StreamCut.UNBOUNDED``` to startStreamCut and en
 current head of stream and the current tail of the stream, respectively.
 
 
-We have provided a simple yet illustrative example of using StreamCut [here.](https://github.com/pravega/pravega-samples/tree/{{variable.pravega_samples_verion}}/pravega-client-examples/src/main/java/io/pravega/example/streamcuts)
+We have provided a simple yet illustrative example of using StreamCut [here.](https://github.com/pravega/pravega-samples/tree/v0.3.0/pravega-client-examples/src/main/java/io/pravega/example/streamcuts)
```
Comment (fpj) [inline documentation/src/docs/streamcuts.md]: I'm confused as of why some pointers point to master while this one points to 0.3.0. That's part of the problem I was pointing out previously with respect to versions. It should point to the version with which the docs are released.

Comment (Prabha-Veerubhotla) [reply documentation/src/docs/streamcuts.md]: There are no pointers to master.  The  pointer is either to `v0.3.0` or github repo `https://github.com/pravega/pravega-samples`.

I forgot to fix only this last time while trying few things.

The next time when we release samples this will be updated accordingly. 
So, now 0.3.0 of pravega -> 0.3.0 of pravega-samples
if we have 0.3.1 of  pravega and no release from samples,  
0.3.1 of pravega -> 0.3.0 of pravega-samples, 

So, no broken links or version mismatch.

File: documentation/src/docs/getting-started.md (hunk 3 above)
Comment (Prabha-Veerubhotla) [reply documentation/src/docs/getting-started.md]: done.

File: documentation/src/docs/getting-started.md (hunk 2 above)
Comment (Prabha-Veerubhotla) [reply documentation/src/docs/getting-started.md]: fixed. it points to `pravega/pravega-samples` now.

File: documentation/src/docs/pravega-concepts.md (hunk 1 above)
Comment (Prabha-Veerubhotla) [reply documentation/src/docs/pravega-concepts.md]: `### AutoScaling: The number of Stream Segments can vary over time `
is referred with the following link

* .md 
`pravega-concepts.md#autoscalingthenumber-of-stream-segments-can-vary-over-time`

* .html
 `pravega-concepts#autoscaling-the-number-of-stream-segments-can-vary-over-time`

In key-features.md, we look for `pravega-concepts.md#autoscalingthenumber-of-stream-segments-can-vary-over-time`,  which is resolved , when we are in  markdown format,

But when we look for the same in `key-features/index.html` , we will unable to resolve
`pravega-concepts#autoscalingthenumber-of-stream-segments-can-vary-over-time`
as that doesn't exist.

To solve that problem i had to drop `The`.

Comment (fpj) [reply documentation/src/docs/pravega-concepts.md]: Change the subtitle then: `Auto-scaling: Varying the number of stream segments over time`.

File: documentation/src/docs/streamcuts.md (hunk 4 above)
Comment (fpj) [reply documentation/src/docs/streamcuts.md]: > The next time when we release samples this will be updated accordingly.

how is this going to be updated?

Comment (Prabha-Veerubhotla) [reply documentation/src/docs/streamcuts.md]: We need to do it manually.  Sorry, it's not automatic.
Initially, we agreed upon, having a  variable in `mkdocs.yml`, which is updated will refer the values in all the `.md` files.  But, this is not possible as of now, because of this : https://github.com/mkdocs/mkdocs/issues/304

File: documentation/src/docs/pravega-concepts.md (hunk 1 above)
Comment (Prabha-Veerubhotla) [reply documentation/src/docs/pravega-concepts.md]: done.

File: documentation/src/docs/streamcuts.md (hunk 4 above)
Comment (fpj) [reply documentation/src/docs/streamcuts.md]: Is there an issue for this? if not, please create it and assign to @nathas1.

Comment (Prabha-Veerubhotla) [reply documentation/src/docs/streamcuts.md]: Created one now https://github.com/pravega/pravega/issues/2767 .
